robottelo/test.py

```diff
@@ -991,3 +987,86 @@ def kick_off_del_test(self, current_num_threads):
             current_num_threads,
             'stat-del-{}-clients'.format(current_num_threads)
         )
+
+    def kick_off_concurrent_sync_test(
+            self,
+            current_num_threads,
+            is_initial_sync):
+        """Refactor out concurrent repository synchronization test case
+
+        :param int current_num_threads: The number of threads
+        :param boolean is_initial_sync: Decide whether resync or initial sync
+        :return dict time_result_dict: Contain a list of X # of timings
+
+        """
+        self.logger.debug(
+            'Concurrent Synchronize by {} threads'
+            .format(current_num_threads)
+        )
+        # determine the target repositories' names by configuration
+        # sync only a sublist of target repos for each test case
+        repo_names_list = self.repo_names_list[:current_num_threads]
+        self.logger.debug(
+            'Targeting repositories to sync: {}'
+            .format(repo_names_list)
+        )
+
+        # Create a list to store all threads
+        thread_list = []
+        # Create a dictionary to store all timing results from each thread
+        time_result_dict = {}
+        for thread_id in range(current_num_threads):
+            time_result_dict['thread-{}'.format(thread_id)] = []
+
+        # sync all specified repositories and repeate X times
+        for iteration in range(self.sync_iterations):
+            # for each thread, sync a single repository
+            for tid in range(current_num_threads):
+                repo_name = repo_names_list[tid]
+                repo_id = self.map_repo_name_id.get(repo_name, None)
+
+                if repo_id is None:
+                    self.logger.warning('Invalid repository name!')
+                    continue
+
+                self.logger.debug(
+                    '{0} repository {1} attempt {2} '
+                    'on {3}-repo test case starts:'
+                    .format(
+                        'Initially sync' if is_initial_sync else 'Resync',
```

Comment:
@Ichimonji10, the `is_initial_sync` is set by: https://github.com/danuzclaudes/robottelo/blob/perf-pulp/tests/foreman/performance/test_pulp_sync.py#L70, therefore a boolean.